src/state_machine.rs

```diff
@@ -123,6 +124,19 @@ impl State {
                 }
             }
             _ => unreachable!(),
+        };
+        info!("Transitioned to: {:?}", &new_self);
```

Comment:
Not sure whether we want to have that user-visible in the vault binaries by default. Maybe it should be `trace!`.